lib/Backend/i386/LowererMDArch.cpp

```diff
@@ -922,6 +922,128 @@ LowererMDArch::LowerWasmMemOp(IR::Instr * instr, IR::Opnd *addrOpnd)
     return doneLabel;
 }
 
+void
+LowererMDArch::LowerAtomicStore(IR::Opnd * dst, IR::Opnd * src1, IR::Instr * insertBeforeInstr)
+{
+    Func* func = insertBeforeInstr->m_func;
+
+    // Move src1 to a register of the same type as dst
+    IR::RegOpnd* tmpSrc = IR::RegOpnd::New(dst->GetType(), func);
+    Lowerer::InsertMove(tmpSrc, src1, insertBeforeInstr);
+    if (dst->IsInt64())
```

Comment:
nit: Maybe put in an assert to make sure that we don't get float dsts, since the spec and this code doesn't support them yet, but not high-priority.

Reply:
done